instrumentation/net/http/otelhttp/handler.go

```diff
@@ -127,8 +127,18 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		trace.WithAttributes(semconv.HTTPServerAttributesFromHTTPRequest(h.operation, "", r)...),
 	}, h.spanStartOptions...) // start with the configured options
 
+	tracer := h.tracer
+
+	if tracer == nil {
+		if span := trace.SpanFromContext(r.Context()); span.SpanContext().IsValid() {
+			tracer = newTracer(span.TracerProvider())
+		} else {
+			tracer = newTracer(otel.GetTracerProvider())
+		}
+	}
```

Comment:
Maybe this code should be moved to some function in the `otel` or package like `otel.TracerProviderFromContext(ctx)`?

Reply:
Not really related to this change as API already supports this behavior. I think I'm personally hesitant of adding new public functions for every case two other public functions are used together. In my codebase, this same check is used a lot of times but there I always return early if `IsValid` is not true as I don't want anything to rely on the global tracer. I guess if this was discussed in the `otel` repo as a shorthand I would find `otel.StartSpan(ctx, name, ..)` more useful for the general case.

Reply:
I'd prefer to avoid adding API surface to the `otel` package if at all possible, particularly while the question of even getting a `TracerProvider` from a `Span` is still outstanding in the spec.  It doesn't seem like there are any particular subtleties to this invocation that would lead me to believe that providing an abstraction is highly valuable.